Plugin.php

```diff
@@ -347,39 +346,35 @@ protected function extendWinterSitemapPlugin(): void
             return;
         }
 
-        $typeMapping = [
-            MLPage::class => ['cms-page'],
-        ];
-
-        if ($pluginManager->exists('Winter.Pages')) {
-            $typeMapping[MLPage::class] = array_merge($typeMapping[MLPage::class], ['static-page', 'all-static-pages']);
-        }
-
-        if ($pluginManager->exists('Winter.Blog')) {
-            $typeMapping[MLBlogCategoryModel::class] = ['blog-category', 'all-blog-categories'];
-            $typeMapping[MLBlogPostModel::class] = ['blog-post', 'category-blog-posts', 'all-blog-posts'];
-        }
-
-        Event::listen('winter.sitemap.processMenuItems', function ($item, $url, $theme, $apiResult) use ($typeMapping) {
-            foreach ($typeMapping as $class => $types) {
-                if (in_array($item->type, $types)) {
-                    return $class::resolveMenuItem($item, $url, $theme);
-                }
+        Event::listen('winter.sitemap.processMenuItems', function ($item, $url, $theme, $apiResult) {
+            if ($item->type === 'cms-page') {
+                return MLPage::resolveMenuItem($item, $url, $theme);
             }
-
             return false;
         });
 
-        Event::listen('winter.sitemap.makeUrlElement',
+        Event::listen('winter.sitemap.addItem',
             function ($definition, $xml, $pageUrl, $lastModified, $itemDefinition, $itemInfo, $itemReference, $urlElement) {
+                $defaultLocale = Locale::getDefault();
```

Comment:
This should probably happen outside of the closure and be passed in via `use ($defaultLocale)`